open-craft/tutor-contrib-superset
# feat: adds superset docker services and open edx integrations

This plugin runs Superset in Tutor, with the following customizations:

* installs python packages to support mysql and OAuth2
* configures SSO from LMS to Superset
* adds custom oauth2 manager which uses LMS APIs and the mysql database to determine user permission levels.
* adds custom course permissions plugin for use with SQL queries.

**Testing instructions**

See updated [README.rst](https://github.com/open-craft/tutor-contrib-superset/tree/jill/install-superset#superset-plugin-for-tutor)

**Author Notes & Concerns**

1. Should we use the openedx redis instance instead of running our own here?
   A: Yes
2. Can/should we use the openedx mysql database for superset instead of running a separate postgres db?
    I tried this, but wasn't able to work out how to get superset's db migrations working.
3. I think this needs a follow-up PR, which updates [superset init script](https://github.com/open-craft/tutor-contrib-superset/blob/jill/install-superset/tutorsuperset/templates/superset/tasks/init-superset.sh) to use the [Superset API](https://superset.apache.org/docs/api) to [create the OpenedX dataset + chart](https://github.com/open-craft/superset/blob/jill/openedx-sso/README-openedx.md#connect-superset-to-the-open-edx-mysql-database), and  [set up roles and row-level security](https://github.com/open-craft/superset/blob/jill/openedx-sso/README-openedx.md#restrict-access-to-open-edx-enrollments-to-course-staff). But I think this can wait until we know exactly how we're going to use Superset with Open edX.
